File: Core/src/main/java/com/github/ezauton/core/localization/sensors/TranslationalDistanceSensor.java
package com.github.ezauton.core.localization.sensors;

/**
 * Like an encoder but for translational distance instead of rotations. An example of a TranslationalDistanceSensor is an {@link EncoderWheel}.
 */
public interface TranslationalDistanceSensor {
    /**
     * Get the translational position
     *
     * @return
     */
    double getPosition();

    /**
     * Get the <strong>translational</strong> velocity (should be VELOCITY so positive <i>and</i> negative)
     *
     * @return
     */
    double getVelocity();
}

File: Core/src/main/kotlin/com/github/ezauton/core/localization/sensors/TranslationalDistanceSensor.kt
package com.github.ezauton.core.localization.sensors

import com.github.ezauton.conversion.Distance
import com.github.ezauton.conversion.LinearVelocity

/**
 * Like an encoder but for translational distance instead of rotations. An example of a TranslationalDistanceSensor is an [EncoderWheel].
 */
interface TranslationalDistanceSensor {
    /**
     * Get the translational position
     *
     * @return
     */
    val position: Distance

    /**
     * Get the **translational** velocity (should be VELOCITY so positive *and* negative)
     *
     * @return
     */
    val velocity: LinearVelocity
}
